blue-green-deploy/blue-green-deploy.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: blue-green-deploy
+spec:
+  workspaces:
+    - name: manifest-dir
+      description: Consisting of kubernetes manifests
+    - name: kubeconfig-dir
+      description: Can be used in case kubeconfig file is mounted
+  params:
+    - name: SERVICE_NAME
+      type: string
+      description: Name of the service which is pointing to existing deployment
+    - name: NEW_VERSION
+      type: string
+      description: The version of newer deployment which is to be patched to existing service
+    - name: MANIFEST
+      type: string
+      default: "."
+      description: The newer version of the deployment to be deployed in another zone
+  steps:
+    - name: change-deployment
+      image: quay.io/openshift/origin-cli:latest
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
+        if [[ -f $(workspaces.kubeconfig-dir.path)/kubeconfig ]]; then
+          export KUBECONFIG=$(workspaces.kubeconfig-dir.path)/kubeconfig
+        fi
+
+        getReadyStatus() {
+          local READY
+          i=0
+          for f in $(kubectl get deploy $1 -o jsonpath='{range .status.conditions[*]}{.reason}{"\n"}{end}')
+          do
+            if [[ $f == "MinimumReplicasAvailable" ]]; then
+              READY=$(kubectl get deploy $1 -o jsonpath='{range .status.conditions['$i']}{.status}{"\n"}{end}')
+              if [[ "$READY" != "True" ]]; then
+                return 1
+              else
+                return 0
+            fi
+            ((i++))
+          done
+          return 1
+        }
```

Comment:
assume the current active deployment is green and the previous deployment was blue.
so if we update blue again, will minimum Replicas consider only upated pods.

in other words does this issue apply here https://github.com/openshift/tektoncd-pipeline-operator/issues/271
and do we need to consider a `getReadyStatus` logic like https://github.com/openshift/tektoncd-pipeline-operator/blob/4ec1bb60bc0f296d57e499c73477b7fe26776967/pkg/utils/validate/validate.go#L35